libs/pika/execution_base/include/pika/execution_base/any_sender.hpp

```diff
@@ -504,15 +504,17 @@ namespace pika::execution::experimental::detail {
         any_receiver& operator=(any_receiver&&) = default;
         any_receiver& operator=(any_receiver const&) = delete;
 
-        friend void tag_invoke(
-            pika::execution::experimental::set_value_t, any_receiver&& r, Ts... ts) noexcept
+        template <typename... Ts_>
+        friend auto tag_invoke(
+            pika::execution::experimental::set_value_t, any_receiver&& r, Ts_&&... ts) noexcept
+            -> decltype(PIKA_MOVE(PIKA_MOVE(r.storage.get())).set_value(PIKA_FORWARD(Ts_, ts)...))
         {
             // We first move the storage to a temporary variable so that
             // this any_receiver is empty after this set_value. Doing
             // PIKA_MOVE(storage.get()).set_value(...) would leave us with a
             // non-empty any_receiver holding a moved-from receiver.
             auto moved_storage = PIKA_MOVE(r.storage);
-            PIKA_MOVE(moved_storage.get()).set_value(PIKA_MOVE(ts)...);
+            PIKA_MOVE(moved_storage.get()).set_value(PIKA_FORWARD(Ts_, ts)...);
```

Comment:
Now that we've moved the possibility of constructors throwing into this function we should add a try-catch block around this, calling `set_error` in the catch.